### PR TITLE
feat: forest-cli snapshot clean

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Implement `forest-cli snapshot clean` to remove all local snapshots as follow up of [this comment](https://github.com/ChainSafe/forest/pull/2039#pullrequestreview-1140887340)

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: forest/cli/src/cli/snapshot_cmd.rs [hunk 1]
```diff
@@ -290,6 +307,36 @@ fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) {
...
+            .clone()
+            .unwrap_or_else(|| default_snapshot_dir(config));
+        println!("Snapshot dir: {}", snapshot_dir.display());
+        let mut snapshots_to_delete = vec![];
+        if let Ok(dir) = fs::read_dir(snapshot_dir) {
+            dir.flatten()
+                .map(|entry| entry.path())
+                .filter(|p| p.extension().unwrap_or_default() == "car")
+                .for_each(|p| snapshots_to_delete.push(p));
+        }
```
Comment (LesnyRumcajs) [inline forest/cli/src/cli/snapshot_cmd.rs:322]: could we `collect` instead? Also, it'd be nice not to omit the error. To my understanding, in the case of a filesystem error (no rights to read for whatever reason, the directory doesn't exist) we would just get `no files to delete`.

It'd be nice to return a `Result` from these commands and provide a juicy error. I assume people using Forest are technical, so even a cryptic IO error should be acceptable.

Comment (hanabi1224) [reply forest/cli/src/cli/snapshot_cmd.rs]: fixed

File: forest/cli/src/cli/snapshot_cmd.rs [hunk 2]
```diff
@@ -290,6 +308,34 @@ fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) {
...
+fn clean(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) -> anyhow::Result<()> {
+    let snapshot_dir = snapshot_dir
+        .clone()
+        .unwrap_or_else(|| default_snapshot_dir(config));
+    println!("Snapshot dir: {}", snapshot_dir.display());
+    let snapshots_to_delete: Vec<_> = fs::read_dir(snapshot_dir)?
+        .flatten()
+        .map(|entry| entry.path())
+        .filter(|p| p.extension().unwrap_or_default() == "car")
+        .collect();
```
Comment (LesnyRumcajs) [inline forest/cli/src/cli/snapshot_cmd.rs:320]: it's beautiful now!

**Review by LesnyRumcajs — APPROVED** [review]

File: forest/cli/tests/snapshot_tests.rs [hunk 3]
```diff
@@ -203,6 +203,78 @@ fn test_snapshot_subcommand_prune_double() -> Result<()> {
...
+        .success();
+
+    let output = std::str::from_utf8(&cmd.get_output().stdout)?.to_owned();
+    for filename in filenames {
+        ensure!(output.contains(filename), output);
+    }
+
+    Ok(())
+}
+
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply forest/cli/tests/snapshot_tests.rs:295]: Fixed.

**Review by lemmih — APPROVED** [review]

**Comment by lemmih** [comment]
I love the cli tests.

**Pull request merged**